diff --git a/updatesrc.sh b/updatesrc.sh
index b977c1d..4a4e83d 100644
--- a/updatesrc.sh
+++ b/updatesrc.sh
@@ -54,7 +54,7 @@ lsfiles () {
 	files sbin/		disklabel dump fsck fsck_ext2fs fsck_ffs
 	files sbin/		fsck_lfs fsck_msdos fsck_v7fs
 	files sbin/		ifconfig mknod
-	files sbin/		modstat mount mount_ffs newfs
+	files sbin/		modstat mount mount_ext2fs mount_ffs newfs
 	files sbin/		newfs_v7fs newfs_msdos newfs_ext2fs
 	files sbin/		newfs_lfs newfs_sysvbfs newfs_udf
 	files sbin/		ping ping6 raidctl reboot
