diff --git a/updatesrc.sh b/updatesrc.sh
index b977c1d..97844ff 100644
--- a/updatesrc.sh
+++ b/updatesrc.sh
@@ -54,7 +54,7 @@ lsfiles () {
 	files sbin/		disklabel dump fsck fsck_ext2fs fsck_ffs
 	files sbin/		fsck_lfs fsck_msdos fsck_v7fs
 	files sbin/		ifconfig mknod
-	files sbin/		modstat mount mount_ffs newfs
+	files sbin/		modstat mount mount_ext2fs mount_ffs newfs
 	files sbin/		newfs_v7fs newfs_msdos newfs_ext2fs
 	files sbin/		newfs_lfs newfs_sysvbfs newfs_udf
 	files sbin/		ping ping6 raidctl reboot
@@ -146,7 +146,7 @@ githubdate ()
 
 	echo '>> final step: commit updatedsrc.sh'
 	cp ../updatesrc.sh .
-	${GIT} commit updatesrc.sh
+	${GIT} commit -m "commit new updatesrc.sh" updatesrc.sh
 
 	echo '>> Done.  Remember to push ./newsrc after testing'
 	echo '>> Use "git diff HEAD^^" to review changes'
